online-installer/install.sh

```diff
@@ -0,0 +1,157 @@
+
+# Making sure important variables exist if not already defined
+#
+# $USER is defined by login(1) which is not always executed (e.g. containers)
+# POSIX: https://pubs.opengroup.org/onlinepubs/009695299/utilities/id.html
+USER="${USER:-$(id -u -n)}"
+# $HOME is defined at the time of login, but it could be unset. If it is unset,
+# a tilde by itself (~) will not be expanded to the current user's home directory.
+# POSIX: https://pubs.opengroup.org/onlinepubs/009696899/basedefs/xbd_chap08.html#tag_08_03
+HOME="${HOME:-$(getent passwd $USER 2> "/dev/null" | cut -d: -f6)}"
+# macOS does not have getent, but this works even if $HOME is unset
+HOME="${HOME:-$(eval echo ~$USER)}"
+
+# Directory to store TPS code repository in
+TPS_LOCAL_REPO="${TPS_LOCAL_REPO:-$HOME/.local/share/tps}"
+
+TPS_REMOTE_REPO_GIT_URL="${TPS_REMOTE_REPO_GIT_URL:-https://github.com/ioi-2017/tps.git}"
+TPS_REMOTE_BRANCH="${TPS_REMOTE_BRANCH:-master}"
+
+function command_exists {
+	command -v "$@" &> "/dev/null"
+}
+
+function fmt_error {
+	printf "Error: %s\n" "$*" >&2
+}
+
+function user_can_sudo {
+	# Checking if sudo is installed
+	command_exists "sudo" || return 1
+	# The following command has 3 parts:
+	#
+	# 1. Runs `sudo` with `-v`. Does the following:
+	#    • with privilege: asks for a password immediately.
+	#    • without privilege: exits with error code 1 and prints the message:
+	#      Sorry, user <username> may not run sudo on <hostname>
+	#
+	# 2. Passes `-n` to `sudo` to tell it to not ask for a password. If the
+	#    password is not required, the command will finish with exit code 0.
+	#    If one is required, sudo will exit with error code 1 and print the
+	#    message:
+	#    sudo: a password is required
+	#
+	# 3. Checks for the words "may not run sudo" in the output to really tell
+	#    whether the user has privileges or not. For that we have to make sure
+	#    to run `sudo` in the default locale (with `LANG=`) so that the message
+	#    stays consistent regardless of the user's locale.
+	#
+	! LANG= sudo -n -v 2>&1 | grep -q "may not run sudo"
+}
+
+
+function clone_tps {
+	# Preventing the cloned repository from having insecure permissions. Failing to do
+	# so causes compinit() calls to fail with "command not found: compdef" errors
+	# for users with insecure umasks (e.g., "002", allowing group writability). Note
+	# that this will be ignored under Cygwin by default, as Windows ACLs take
+	# precedence over umasks except for filesystems mounted with option "noacl".
+	umask g-w,o-w
+
+	echo -n " ########################## "
+	echo -n "Cloning TPS Repo to ${TPS_LOCAL_REPO}..."
+	echo -n " ########################## "
+	echo
+
+	command_exists "git" || {
+		fmt_error "git is not installed"
+		exit 1
+	}
+
+	ostype=$(uname)
+	if [ -z "${ostype%CYGWIN*}" ] && git --version | grep -q msysgit; then
+		fmt_error "Windows/MSYS Git is not supported on Cygwin." \
+			"Make sure the Cygwin git package is installed and is first on the \$PATH"
```

Comment:
```suggestion
		fmt_error "\
Windows/MSYS Git is not supported on Cygwin.
Make sure the Cygwin git package is installed and is first on the \$PATH."
```

Reply:
I think it's better for the error to be in one line.